go/pkg/apis/recognizer/component_recognizer.go

```diff
@@ -209,34 +195,66 @@ func isFirstPathParentOfSecond(firstPath string, secondPath string) bool {
 	Returns:
 		list of components detected or err if any error occurs
 */
-func DetectComponentsFromFilesList(files []string, settings model.DetectionSettings) ([]model.Component, error) {
+func DetectComponentsFromFilesList(files []string, settings model.DetectionSettings) []model.Component {
 	configurationPerLanguage := langfiles.Get().GetConfigurationPerLanguageMapping()
 	var components []model.Component
 	for _, file := range files {
-		dir, fileName := filepath.Split(file)
-		if dir == "" {
-			dir = "./"
+		languages, err := getLanguagesByConfigurationFile(configurationPerLanguage, file)
+		if err != nil {
+			continue
 		}
-		languages, err := getLanguagesByConfigurationFile(configurationPerLanguage, fileName)
+		// check if possible to pick only one lang (e.g tsconfig.json or jsconfig.json)
+		languages = narrowLanguagesList(languages, file)
+
+		component, err := detectComponentUsingConfigFile(file, languages, settings)
 		if err != nil {
 			continue
 		}
-		for _, language := range languages {
-			if isConfigurationValid(language, file) {
-				component, _ := detectComponent(dir, languages, settings)
-				if component.Path != "" {
-					components = appendIfMissing(components, component)
-					break
-				}
+		components = appendIfMissing(components, component)
+	}
+	return components
+}
+
+func narrowLanguagesList(languages []string, file string) []string {
+	if len(languages) == 1 {
+		return languages
+	}
+
+	dir, _ := utils.NormalizeSplit(file)
+	langSize := len(languages) - 1
+	for index, language := range languages {
+		if !isLanguageValid(language, dir) {
+			if langSize == index {
+				languages = languages[:index]
+			} else {
+				languages = append(languages[:index], languages[index+1:]...)
 			}
 		}
 	}
-	return components, nil
+
+	return languages
+}
+
+func isLanguageValid(language string, dirPath string) bool {
+	configFile := ""
+	if strings.EqualFold(language, "javascript") {
+		configFile = "jsconfig.json"
+	} else if strings.EqualFold(language, "typescript") {
+		configFile = "tsconfig.json"
+	}
+
+	if configFile == "" {
+		return true
+	}
+	if _, err := os.Stat(filepath.Join(dirPath, configFile)); os.IsNotExist(err) {
```

Comment:
Not sure to fully understand: does it mean those files are required to a Javascript or Typescript component.

Reply:
The idea was that one but after reading your comments and a bit of investigation i found out those files are not mandatory, so now i check if the typescript dependency is in the project, if so the main lang is typescript otherwise javascript